text/0000-partial-template.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Partial Template
+status: DRAFTED
+created_at: 2020-01-16
+updated_at: 2020-01-17
+pr: (leave this empty until the PR is created)
+---
+
+# Short Title
+
+## Summary
+
+Partial templates allow the ability to import html templates into another template via
+the normal variable syntax `{template}`.
+
+## Basic example
+
+```js
+import partialTemplate from './partialTemplate.html';
+
+export default class Example extends LightningElement {
+    partialTemplate = partialTemplate;
+}
+```
+
+```html
+<template>
+  Parent
+  {partialTemplate}
+</template>
+```
+
+## Motivation
+
+In investigating performance issues in components that are used hundreds of times on a
+page it was noted that creating reusable child components; while organizationally nice,
+were not as performant as inlining the HTML into the template;
+
+In other instances it was noted that large conditonal blocks of HTML could be more organized
+into seperate template files.
+
+## Detailed design
+
+Templates can be rendered into a component one of two ways currently.
+
+- By matching the name of the component. Ex: `cmp.js` / `cmp.html`
+- By returning a default import in the `render() {}` method.
+  - Ex: `import cmpTemplate from './cmp.html'` -> `render() { return cmpTemplate; }`
+
+A template is a constant and cannot be modified. Partials will inherit this same behavior.
+
+This simplifies things, but still may lead to complications in optimizations [unfamiliar with
+what this entails, so may need LWC context here]. Example:
+
+```
+get dynamicTemplate() {
+    return this.conditonal ? template1 : template2;
+}
+```
+
+## Drawbacks

```

Comment:
one big drawback here is the static analysis... as today, analyzing the template can give us a lot of information, but with something like this, it makes it harder